docs/configuring-crash-stats.rst

```diff
@@ -0,0 +1,310 @@
+.. index:: configuring-crashstats
+
+.. _configuring-crashstats-chapter:
+
+Configuring Crash-Stats and PostgreSQL
+======================================
+
+Install PostgreSQL
+------------------
+
+Install the PostgreSQL repository.
+::
+  sudo rpm -ivh http://yum.postgresql.org/9.3/redhat/rhel-7-x86_64/pgdg-centos93-9.3-1.noarch.rpm
+
+Now you can actually install the packages:
+::
+  sudo yum install postgresql93-server postgresql93-contrib
+
+Initialize and enable PostgreSQL on startup:
+::
+  sudo service postgresql-9.3 initdb
+  sudo systemctl enable postgresql-9.3
+
+Modify postgresql config
+::
+  sudo vi /var/lib/pgsql/9.3/data/postgresql.conf
+
+Ensure that timezone is set to UTC
+::
+  timezone = 'UTC'
+
+Allow local connections for PostgreSQL
+::
+  sudo vi /var/lib/pgsql/9.3/data/pg_hba.conf
+
+Ensure that local connections are allowed:
+::
+
+  # IPv4 local connections:
+  host    all             all             127.0.0.1/32            trust
+  # IPv6 local connections:
+  host    all             all             ::1/128                 trust
+
+See http://www.postgresql.org/docs/9.3/static/auth-pg-hba-conf.html
+for more information on this file.
+
+You'll need to restart postgresql if the configuration was updated:
+::
+  sudo systemctl restart postgresql-9.3
+
+Create database, set up schema
+------------------------------
+
+Socorro provides a setup script which attempts to initialize PostgreSQL::
+
+    sudo setup-socorro.sh postgresql
+
+This creates a new database named "breakpad" and sets up the schema
+that Socorro and crash-stats expect.
+
+Configure Socorro Processor
+---------------------------
+
+::
+  socorro processor \
+    --destination.crashstorage_class='
+      socorro.external.crashstorage_base.PolyCrashStorage' \
+    --destination.storage_classes='
+      socorro.external.fs.crashstorage.FSLegacyDatedRadixTreeStorage,       
+      socorro.external.postgresql.crashstorage.PostgreSQLCrashStorage'
+
+
+Start services
+--------------
+
+Both the Django socorro-webapp and the socorro-middleware REST service
+must be running::
+
+    sudo systemctl enable socorro-middleware socorro-webapp
+
+Configure Nginx
+---------------
+
+Both socorro-webapp and socorro-middleware should be fronted by a
+webserver like Nginx.
```

Comment:
I think the easy answers are because Nginx listens on port 80 and can handle static resources more efficiently. 

Nginx also buffers slow clients which I know can easily lead to a DoS with gunicorn at least (described http://gunicorn-docs.readthedocs.org/en/latest/deploy.html), uwsgi's direct HTTP serving feature may be able to handle this idk.